core/server/api/v2/slugs.js

```diff
@@ -0,0 +1,39 @@
+const models = require('../../models');
+const common = require('../../lib/common');
+
+const allowedTypes = {
+    post: models.Post,
+    tag: models.Tag,
+    user: models.User,
+    app: models.App
+};
+
+module.exports = {
+    docName: 'slugs',
+    generate: {
+        options: [
+            'include',
+            'type'
+        ],
+        data: [
+            'name'
+        ],
+        permissions: true,
+        validation: {
+            options: {
+                type: Object.keys(allowedTypes)
+            }
+        },
+        query(frame) {
+            return models.Base.Model.generateSlug(allowedTypes[frame.options.type], frame.data.name, {status: 'all'})
```

Comment:
Looks like `data.name` is required?

Reply:
Good catch 🎣 Updated `required` for both `data.name` and `options.type`.